include/zephyr/net/wifi_mgmt.h

```diff
@@ -136,20 +168,21 @@ struct net_wifi_mgmt_offload {
 	int (*ap_enable)(const struct device *dev,
 			 struct wifi_connect_req_params *params);
 	int (*ap_disable)(const struct device *dev);
+	struct wifi_iface_status *(*iface_status)(const struct device *dev);
+#ifdef CONFIG_NET_STATISTICS_WIFI
+	struct net_stats_wifi *(*get_stats)(const struct device *dev);
+#endif
 };
 
 /* Make sure that the network interface API is properly setup inside
  * Wifi mgmt offload API struct (it is the first one).
  */
 BUILD_ASSERT(offsetof(struct net_wifi_mgmt_offload, iface_api) == 0);
 
-#ifdef CONFIG_WIFI_OFFLOAD
-
 void wifi_mgmt_raise_connect_result_event(struct net_if *iface, int status);
 void wifi_mgmt_raise_disconnect_result_event(struct net_if *iface, int status);
-
-#endif /* CONFIG_WIFI_OFFLOAD */
-
+void wifi_mgmt_raise_iface_status_event(struct net_if *iface,
```

Comment:
@krish2718 Is there an overlap between `wifi_mgmt_raise_iface_status_event` and functions defined above? In other words are these more specific functions to `wifi_mgmt_raise_iface_status_event`?

If so perhaps we should dump these two and just keep `wifi_mgmt_raise_iface_status_event`?

Reply:
There is a slight overlap, as `status` is a superset, but as these are fundamental operations keeping specific events is helpful than parsing status and figuring out.